cadence/contracts/interfaces/DeFiActions.cdc

```diff
@@ -895,12 +980,222 @@ access(all) contract DeFiActions {
+            for id in self._scheduledTransactions.keys {
+                if id == whileExecuting {
+                    continue
+                }
+                let scheduledTxn = self.borrowScheduledTransaction(id: id)!
+                if scheduledTxn.status() == FlowTransactionScheduler.Status.Scheduled {
+                    // found another scheduled transaction within the configured interval
+                    if scheduledTxn.timestamp <= timestamp! {
+                        return nil
+                    }
+                }
+            }
+
+            // fallback in event there was an issue with assigning the last rebalance timestamp or last rebalance was
+            // executed long ago
+            let config = self._recurringConfig!
+            let now = getCurrentBlock().timestamp
+            if timestamp! < now {
+                // protect overflow & update timestamp value
+                if UInt64(UFix64.max) - UInt64(now) < UInt64(config.interval) {
+                    return "INTERVAL_OVERFLOW"
+                }
+                timestamp = now + UFix64(config.interval)
+            }
+
+            // estimate the transaction fees
+            let estimate = FlowTransactionScheduler.estimate(
+                data: config.forceRebalance,
+                timestamp: timestamp!,
+                priority: config.priority,
+                executionEffort: config.executionEffort
+            )
+            // post-estimate check if the estimate is valid & that the funder has enough funds of the correct type
+            // NOTE: low priority estimates always receive non-nil errors but are still valid if fee is also non-nil
+            if (estimate.flowFee == nil && estimate.error != nil)
+                || config.txnFunder.minimumAvailable() < estimate.flowFee!
+                || config.txnFunder.getSourceType() != Type<@FlowToken.Vault>() {
+                var errorMessage = estimate.error!
+                if config.txnFunder.getSourceType() != Type<@FlowToken.Vault>() {
+                    errorMessage = "INVALID_FEE_TYPE"
+                } else if config.txnFunder.minimumAvailable() < estimate.flowFee! {
+                    errorMessage = "INSUFFICIENT_FEES_AVAILABLE"
+                }
+                return errorMessage
+            }
+
+            // withdraw the fees from the funder & post-withdraw check that the provided fees are sufficient
+            let fees <- config.txnFunder.withdrawAvailable(maxAmount: estimate.flowFee!) as! @FlowToken.Vault
+            if fees.balance < estimate.flowFee! {
+                config.txnFunder.depositCapacity(from: &fees as auth(FungibleToken.Withdraw) &{FungibleToken.Vault})
+                destroy fees
+                return "INSUFFICIENT_FEES_PROVIDED"
+            } else {
+                // all checks passed - schedule the transaction & capture the scheduled transaction
+                let txn <- FlowTransactionScheduler.schedule(
```

Comment:
The whole selfCap property and setter can be removed in favor of using manager's **scheduleByHandler** function because the handler has been already scheduled at least 1 time here and its capability is already inside the manager (https://github.com/onflow/flow-core-contracts/blob/aab7a9dda680bbac49b1d12f0f69b18e8490a943/contracts/FlowTransactionSchedulerUtils.cdc#L168)